tests/unit/test_allocator.cpp

```diff
@@ -0,0 +1,182 @@
+#include "gtest/gtest.h"
+#include "VecSim/vec_sim.h"
+#include "VecSim/memory/vecsim_malloc.h"
+#include "VecSim/memory/vecsim_base.h"
+#include "VecSim/algorithms/brute_force/brute_force.h"
+#include "VecSim/algorithms/hnsw/hnswlib_c.h"
+
+class AllocatorTest : public ::testing::Test {
+protected:
+    AllocatorTest() {}
+
+    ~AllocatorTest() override {}
+
+    void SetUp() override {}
+
+    void TearDown() override {}
+};
+
+struct SimpleObject : public VecsimBaseObject {
+public:
+    SimpleObject(VecSimAllocator *allocator) : VecsimBaseObject(allocator) {}
+    SimpleObject(std::shared_ptr<VecSimAllocator> allocator) : VecsimBaseObject(allocator) {}
+    int x;
+};
+
+struct ObjectWithSTL : public VecsimBaseObject {
+    std::vector<int, VecsimSTLAllocator<int>> test_vec;
+
+public:
+    ObjectWithSTL(VecSimAllocator *allocator) : VecsimBaseObject(allocator), test_vec(allocator){};
+    ObjectWithSTL(std::shared_ptr<VecSimAllocator> allocator)
+        : VecsimBaseObject(allocator), test_vec(allocator){};
+};
+
+struct NestedObject : public VecsimBaseObject {
+    ObjectWithSTL stl_object;
+    SimpleObject simpleObject;
+
+public:
+    NestedObject(VecSimAllocator *allocator)
+        : VecsimBaseObject(allocator), stl_object(allocator), simpleObject(allocator){};
+    NestedObject(std::shared_ptr<VecSimAllocator> allocator)
+        : VecsimBaseObject(allocator), stl_object(allocator), simpleObject(allocator){};
+};
+
+TEST_F(AllocatorTest, test_simple_object) {
+    std::shared_ptr<VecSimAllocator> allocator = std::make_shared<VecSimAllocator>();
+    SimpleObject *obj = new (allocator) SimpleObject(allocator);
+    ASSERT_GE(allocator->getAllocationSize(), sizeof(SimpleObject) + sizeof(VecSimAllocator));
```

Comment:
Why `ASSERT_GE`? Can't we know the expected allocation exactly?

Reply:
in this case you comment is right, good catch
however when using STL, this is not always hold as STL implementation should be a "black box" and we cannot assume anything regarding exact allocation